accessible/workspace_treeview.component.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * AccessibleBlockly
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how Blockly.Block's are
+ * rendered in the workspace in AccessibleBlockly. Also handles any
+ * interactions with the blocks.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+blocklyApp.WorkspaceTreeView = ng.core
+  .Component({
+    selector: 'tree-view',
+    template: `
+      <li #parentList [id]="idMap['parentList']" role="treeitem" class="blocklyHasChildren" 
+          [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr('blockly-block-summary', idMap['blockSummary'])" 
+          [attr.aria-level]="level" aria-selected=false>
+        {{checkParentList(parentList)}}
```

Comment:
Why is this here? Same concerns as before. Can it be executed from JS on, e.g., init?